packages/libs/client-utils/src/nodejs/deploy-from-directory.ts

```diff
@@ -0,0 +1,194 @@
+ * @param clientConfig - The client configuration for the deployment
+ * @throws - If no template or code files are provided
+ * @example
+ * ```typescript
+ * await deployFromDirectory({
+ *    chainIds: [chainId],
+ *    templateConfig: {
+ *      path: 'path',
+ *      extension: 'yaml',
+ *      sort: () => 1,
+ *      namespaceExtractor: () => 'namespace',
+ *      deploymentArguments: {
+ *        chain: chainId,
+ *      },
+ *    },
+ *    clientConfig,
+ * });
+ * ```
+ *
+ */
+export const deployFromDirectory = async ({
+  chainIds,
+  templateConfig,
+  codeFileConfig,
+  clientConfig,
+}: IDeployConfig) => {
+  if (!templateConfig && !codeFileConfig) {
+    throw new Error('Please provide a template or a code files directory');
+  }
+
+  let templateFiles: Array<string>;
+  let codeFiles: Array<string>;
+
+  if (templateConfig) {
+    templateFiles = readdirSync(templateConfig!.path).filter((file) =>
+      file.endsWith(templateConfig!.extension),
+    );
+
+    if (templateConfig.sort) {
+      templateFiles.sort(templateConfig.sort);
+    }
+  } else if (codeFileConfig) {
+    codeFiles = readdirSync(codeFileConfig!.path).filter((file) =>
+      file.endsWith(codeFileConfig!.extension),
+    );
+
+    if (codeFileConfig.sort) {
+      codeFiles.sort(codeFileConfig.sort);
+    }
+  }
+
+  await Promise.all(
+    chainIds.map(async (chainId) => {
+      // If there are template files, we'll build the pact commands from them
+      if (templateFiles && templateFiles.length > 0) {
+        for (const templateFile of templateFiles) {
+          try {
+            const deployArguments = templateConfig?.deploymentArguments || {};
+
+            if (templateConfig?.namespaceExtractor) {
+              const namespace = templateConfig.namespaceExtractor(templateFile);
+              deployArguments.namespace = namespace;
+            }
+
+            deployArguments.chain = chainId;
+
+            const command = await deployTemplate(
+              {
+                deployArguments,
+                workingDirectory: templateConfig?.path,
+                templatePath: templateFile,
+              },
+              clientConfig,
+            );
+
+            const commandResult = await command.executeTo('listen');
+
+            if (commandResult.result.status === 'success') {
+              console.log(
+                `Successfully deployed ${templateFile} on chain ${chainId}`,
+              );
+            } else {
+              console.log(
+                `Failed to deploy ${templateFile} on chain ${chainId}. Error: ${commandResult.result.error}`,
+              );
+            }
+          } catch (error) {
+            console.error(error);
+          }
+        }
+      } else if (
+        codeFiles &&
+        codeFiles.length > 0 &&
+        codeFileConfig &&
+        !templateConfig
+      ) {
```

Comment:
Might be good to create two separate functions. One for the first block, and one for the second block. Maybe even move the choice for which one to use up the stack (to the caller code).

Given these two statements: 
```
      if (templateFiles && templateFiles.length > 0) {
```
and

```ts
      if (
        codeFiles &&
        codeFiles.length > 0 &&
        codeFileConfig &&
        !templateConfig
      )
```